### ISSUE TITLE
How to list only opened files when switching buffers

### ISSUE THREAD

**Issue by c02y** [abo-abo/swiper#2896]
When I switch buffers using `ivy-switch-buffer` or `counsel-ibuffer` command, most likely I need to switch to an opened files, but when I use Emacs for long enough, there are a lot of buffers opened like `*lsp-xxx*`, `*ccls-xxx*`, I don't need anything like `*xxx*` in the list most of the time, so if I open a new instance of Emacs, I use the switch-buffer(`ivy-switch-buffer` or `counsel-ibuffer`) command, the list will be empty, even the `*scratch*` or `*Message*` will not be there. How can do that?

BTW: one command for the full list of buffers should be provided.

**Comment by basil-conto** [comment]
Have you tried customising `ivy-ignore-buffers` and `ivy-use-ignore-default`?

**Comment by c02y** [comment]
I found 
```
(setq ivy-ignore-buffers '("\\` " "\\`\\*"))
```
from https://github.com/abo-abo/swiper/issues/644#issuecomment-243478567

But cannot make it exactly as what I need.

1. Personally, I like to use `counsel-ibuffer` since it searches from the path as well.
2. `ivy-ignore-buffers` doesn't affect `counsel-ibuffer`(it affects `counsel-switch-buffer` though, I thought ivy and counsel are independent)
3. I need a command to list all the buffers and virtual-buffers(`ivy-use-virtual-buffers`)

So basically I need a variable like `ivy-ignore-buffers` but for `counsel-ibuffer`, then I don't need to change `ivy-use-virtual-buffers` and use `counsel-ibuffer` for the purpose of switch opened buffers and use `ivy-switch-ibuffer` for the full list of buffers+virtual-buffers.

**Comment by basil-conto** [comment]
> I thought ivy and counsel are independent

Perhaps that was the original intention in theory, but in practice they're coupled as part of the same package. Ivy provides the completion logic, and Counsel provides utilities using Ivy's features. The same applies for Swiper.  So customising Ivy in general affects all other packages using it (although this isn't 100% consistent).

> So basically I need a variable like ivy-ignore-buffers but for counsel-ibuffer

This may be possible by passing a `qualifiers` argument to `ibuffer` in `counsel-ibuffer--get-buffers`.  Patches welcome, and TIA.
